### PR TITLE
Migrate to Material 3

### LINKED ISSUE #107: Migrate to Material 3
- [x] [compose-material-dialogs](https://github.com/PranavMaganti/compose-material-dialogs/tree/main) uses M2, and seems that is no longer maintained, so we ~might~ **want** to move away from it. 
  - [X] Follow [Migration guide](https://developer.android.com/jetpack/compose/designsystems/material2-material3)
  - [x] Fix light theme colors
  - [x] Fix dark theme colors 
  - [x] use M3's [`supportingText` param on textfields](https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#textfield) to show examples or hints like "Field is required" and error messages for invalid input 

_Note:_ ~We can't get rid of the M2 dependency yet, as pullRefresh (new name [pullToRefresh](https://developer.android.com/reference/kotlin/androidx/compose/material3/pulltorefresh/package-summary)) has not been migrated yet. Its in beta already though.~

### PR DESCRIPTION
**Pull request by sunkup**
@ArnyminerZ 

I could use some help on the theming. I am having a really hard time figuring out how to set the colors in the theme correctly.  I would like to leave the composables as they are and not hard code anything. It should be possible with this fancy new theming engine ... I hope ?

Styling the TopAppBar with colors also changes properties of textfields and I would like the dialog boxes to be white or offwhite, instead of the default generated blueish hue, but that seems impossible - alternative would be to use keppelers "info" and "select" dependency modules for our other dialog boxes too. They have very nice defaults which look great IMO.  

See the comment [here](https://github.com/ArnyminerZ/icsx5/pull/109/commits/6b8c1305548e6b5235e1d66fa201a7b266537c4f#diff-f0e4c12957ab2e61fcacb3486d2bb944927ae6a0b8786489c89852bc301f1a0eR49-R58) for details

On a side note: I know ICSx5 had the switch colors as red, but I think that's really bad UX, so I think the primary color (lightblue) is just fine here.

### CODE REVIEW

**Comment by ArnyminerZ** [comment]
> I am having a really hard time figuring out how to set the colors in the theme correctly

I doubt it's possible to set everything from the theme. Also there are styling differences between M2 and M3 regarding top bars ([see](https://m3.material.io/components/top-app-bar/overview#6414ed19-c273-4f32-94d5-5ef782303023)).

I think the idea is to set some "general colors" in the theme, and then, if you want to customize something better, use the colors for that specific component (this can be defined somewhere and kept common), or use [CompositionLocal](https://developer.android.com/jetpack/compose/compositionlocal) for smaller adjustments.

Note that [even in the guidelines](https://m3.material.io/components/menus/overview), the top app bar and the dropdowns have the same color, so I think it's the way to go. However, I don't like the dropdowns being that much blue.

> Styling the TopAppBar with colors also changes properties of textfields and I would like the dialog boxes to be white or offwhite, instead of the default generated blueish hue

Then I believe the way to go is to set custom colors for the action bar, and leave the rest as default as possible.

**Comment by sunkup** [comment]
> Then I believe the way to go is to set custom colors for the action bar, and leave the rest as default as possible.

Darn, ok. Will try do something sensible

> Note that [even in the guidelines](https://m3.material.io/components/menus/overview), the top app bar and the dropdowns have the same color, so I think it's the way to go. However, I don't like the dropdowns being that much blue.

Well, we have a clear goal. Since this is a refactor, for now, everything should look (as much as possible) the same as before. :)

**Comment by ArnyminerZ** [comment]
> everything should look (as much as possible) the same as before

The thing is that migrating to M3 implicates a design change, it's not a framework update but an stylistic update. So we really don't want the same design, we want to use the new design and guidelines. If we want to keep the same design as before, we should just stick with M2.

**Comment by sunkup** [comment]
> The thing is that migrating to M3 implicates a design change, it's not a framework update but an stylistic update. So we really don't want the same design, we want to use the new design and guidelines. If we want to keep the same design as before, we should just stick with M2.

Oh, I guess you are right about that. Do we know whether Ricki and Bernhard are interested in M3 then? If not, we should drop the whole M3 thing.

Although I think it looks quite similar to the way it looked before right now ... Apart from the blue switches, which we could turn red again as well, if they might want that ...

**Comment by ArnyminerZ** [comment]
> Do we know whether Ricki and Bernhard are interested in M3 then?

Yup, we should ask them first. I think they want to, since Ricki mentioned M3 migration also for DAVx5 at some point, but yeah, it will be a redesign, so we should ask. Maybe they want more control, and do it once we merge into the official repo.

**Comment by sunkup** [comment]
There are still minor style issues, which I don't want to work on now, as Ricki and Bernhards might want it differently. So I feel we should make those changes in the main repo too.

File: app/src/main/java/at/bitfire/icsdroid/ui/theme/Theme.kt [hunk 1]
```diff
@@ -10,56 +10,74 @@ import androidx.compose.foundation.isSystemInDarkTheme
...
     darkTheme: Boolean = isSystemInDarkTheme(),
     content: @Composable () -> Unit
 ) {
     val context = LocalContext.current
 
-    val colors = if (darkTheme)
+    val colorScheme = if (darkTheme) {
         DarkColors
-    else
+    } else
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/theme/Theme.kt]: I'd also add `{}` here, or remove the other one. Just an stylistic thing.

**Review by ArnyminerZ — APPROVED** [review]
Just a little comment, a stupid thing. Otherwise everything looks fine and seems to work fine for me 😄

**Pull request merged**